R/produce_source_sc_tests.R

```diff
@@ -0,0 +1,51 @@
+#' Source Care Home Tests
+#'
+#' @description Produce a set of tests which can be used by Care Home and Home Care
+#' This will produce counts of various demographics
+#' using [create_demog_test_flags()] counts of episodes for every `hbtreatcode`
+#' using [create_lca_test_flags()]
+#' It will also produce various summary statistics for bedday, cost and
+#' episode date variables.
+#'
+#' @param data new or old data for testing summary flags
+#' (data is from [get_source_extract_path()])
+#' @param sum_mean_vars variables used when selecting 'all' measures from [calculate_measures()]
+#' @param max_min_vars variables used when selecting 'min-max' from [calculate_measures()]
+#'
+#' @return a dataframe with a count of each flag
+#' from [calculate_measures()]
+#' @export
+#'
+#' @family extract test functions
+#' @seealso calculate_measures
+produce_source_sc_tests <- function(data,
+                                    sum_mean_vars = c("beddays", "cost", "yearstay"),
+                                    max_min_vars = c(
+                                      "record_keydate1", "record_keydate2",
+                                      "cost_total_net", "yearstay"
+                                    )) {
+  test_flags <- data %>%
+    # use functions to create HB and partnership flags
+    create_demog_test_flags() %>%
+    dplyr::mutate(n_episodes = 1) %>%
```

Comment:
```suggestion
        dplyr::mutate(n_episodes = 1,
                  hc_per = dplyr::if_else(.data$smrtype == "HC-Per", 1, 0),
                  hc_non_per = dplyr::if_else(.data$smrtype == "HC-Non-Per", 1, 0),
                  hc_unknown = dplyr::if_else(.data$smrtype == "HC-Unknown",1, 0),
                  hc_reablement_no = dplyr::if_else(.data$smrtype == 0, 1, 0),
                  hc_reablement_yes = dplyr::if_else(.data$smrtype == 1, 1, 0),
                  hc_reablement_unknown = dplyr::if_else(.data$smrtype == 9, 1, 0)
                  )
    ) %>%
```
Some variables not picked up by the function - this should work if you include these here